contracts/FlashRollover.sol

```diff
@@ -254,24 +243,38 @@ contract FlashRollover is IFlashRollover {
     }
 
     function _validateRollover(
-        LoanLibrary.LoanData memory loanData,
-        ILoanCore loanCore,
+        ILoanCore sourceLoanCore,
         ILoanCore targetLoanCore,
-        LoanLibrary.LoanTerms calldata newLoanTerms
+        LoanLibrary.LoanTerms memory sourceLoanTerms,
+        LoanLibrary.LoanTerms calldata newLoanTerms,
+        uint256 borrowerNoteId
     ) internal {
-        uint256 borrowerNoteId = loanData.borrowerNoteId;
+        require(sourceLoanCore.borrowerNote().ownerOf(borrowerNoteId) == msg.sender, "caller not borrower");
 
-        IERC721 borrowerNote = loanCore.borrowerNote();
-        address borrower = borrowerNote.ownerOf(borrowerNoteId);
-        require(borrower == msg.sender, "Rollover: borrower only");
+        require(newLoanTerms.payableCurrency == sourceLoanTerms.payableCurrency, "currency mismatch");
 
-        LoanLibrary.LoanTerms memory terms = loanData.terms;
+        require(newLoanTerms.collateralTokenId == sourceLoanTerms.collateralTokenId, "collateral mismatch");
 
-        require(newLoanTerms.payableCurrency == terms.payableCurrency, "Currency mismatch");
-        require(newLoanTerms.collateralTokenId == terms.collateralTokenId, "Collateral mismatch");
         require(
-            address(loanCore.collateralToken()) == address(targetLoanCore.collateralToken()),
-            "Non-compatible AssetWrapper"
+            address(sourceLoanCore.collateralToken()) == address(targetLoanCore.collateralToken()),
+            "non-compatible AssetWrapper"
         );
     }
+
+    function setOwner(address _owner) external override {
```

Comment:
why not just use oz ownable?

Reply:
Cost 🤷  I figured the usage surface area is so small, we didn't need the full functionality of `Ownable` and could save a bit on deployment and usage - wouldn't want things like `renounceOwnership` to work in this case anyway